### ISSUE TITLE
throw ArrayIndexOutOfBoundException When loading map 

### ISSUE THREAD

**Issue by probepark** [l1j/en#305]
Loading Map...
Maps 547
2015.03.22 20:53:39 L1WorldMap.<init> SEVERE: 192
2015.03.22 20:53:39 java.lang.ArrayIndexOutOfBoundsException: 192
	at l1j.server.server.model.map.TextMapReader.read(TextMapReader.java:88)
	at l1j.server.server.model.map.TextMapReader.read(TextMapReader.java:114)
	at l1j.server.server.model.map.CachedMapReader.cacheMap(CachedMapReader.java:70)
	at l1j.server.server.model.map.CachedMapReader.read(CachedMapReader.java:92)
	at l1j.server.server.model.map.CachedMapReader.read(CachedMapReader.java:127)
	at l1j.server.server.model.map.L1WorldMap.<init>(L1WorldMap.java:45)
	at l1j.server.server.model.map.L1WorldMap.getInstance(L1WorldMap.java:35)
	at l1j.server.server.GameServerThread.<init>(GameServerThread.java:170)
	at l1j.server.server.GameServerThread.getInstance(GameServerThread.java:88)
	at l1j.server.server.GameServer.initialize(GameServer.java:127)
	at l1j.server.Server.main(Server.java:66)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)

**Comment by Zylaxyma** [comment]
It appears your system is referencing mapid 547, our database does not currently contain entries for this map, and I was unable to locate any other references that call upon this mapid. Please review your code version to see why this entry exists where it is not listed in the base code

**Comment by asung** [comment]
I've got the exact same issue with current master branch.
Built with ant script, and I only ran one sql script (db/l1jdb_m8.sql) before try running the server.
When I start server, I get this while server is loading maps.
I haven't debug much yet, but I think reproduce step is probably just to fresh checkout and setup using current master branch.

**Comment by asung** [comment]
Have not found fix for it yet as I am not familiar with the code base and data. However, I think the "Maps 547" does not mean mapid 547. It just logs that there are 547 map files in maps folder that it is loading.

Maybe, one of the map file has invalid data.

**Comment by asung** [comment]
More findings after adding some debug statements and run it.

At least for me, problem is with map 72.txt file. TextMapReader.java gets map size (width and height) as xSize and ySize. These comes from mapids table. Based on the table, xSize and ySize of the map should be 128 by 128. However, actual map file has data 192 by 128. 

Therefore, it allocates 128x128 array and then try to put bigger data and cause ArrayIndexOutOfBoundsException.

**Comment by asung** [comment]
I was able to run the server after I modified some data on mapids table.
Mapids with wrong sizes were 72, 73, 74, and 2000. I either added values to endX/endY  or subtracted values to startX/startY. Not sure if these values will work well with client.

As I do not know the implication of changing x,y coordinate on these tables, I would ask someone else to make proper changes to fix this.

FYI, these are the new values I have on mapids table.
72	Ice Crystal Cave	32704	32895	32768	32895	1	1	0	0	0	1	1	1	1	1	1	1	1
73	Ice Queens Castle 2F	32704	32895	32768	32959	1	1	0	0	0	1	1	1	1	1	1	1	1
74	Ice Queens Castle 3F	32704	32895	32768	32959	1	1	0	0	0	1	1	1	1	1	1	1	1
2000	Ancient Space 1F	32704	33023	32704	33023	1	1	0	0	0	1	1	1	1	1	1	1	1
